tests/test_lauch_states.py

```diff
@@ -0,0 +1,217 @@
+    @patch('util.config_life_cycle.ConfigParser.load_file', MagicMock(return_value=parsed_config))
+    @patch('util.config_life_cycle.ConfigLifeCycle.get_baking_cfg_file', MagicMock(return_value=""))
+    def test_wrong_node_end_point(self):
+        # Test with PRPC node
+        args = Args(initial_cycle=10, reward_data_provider='tzkt', api_base_url='https://api.carthage.tzkt.io/v1/')
+        args.node_endpoint = 'https://testnet-tezos.giganode.io:4433'
+        args.docker = True
+        args.dry_run = True
+        args.dry_run_no_consumers = True
+        args.syslog = False
+        args.verbose = "off"
+        args.log_file = 'logs/app.log'
+        args.do_not_publish_stats = True
+        args.run_mode = 3
+
+        start_application(args)
+
+    @patch('log_config.main_logger', test_logger)
+    @patch('cli.client_manager.ClientManager.check_pkh_known_by_signer', MagicMock(return_value=True))
+    @patch('cli.client_manager.ClientManager.get_bootstrapped', MagicMock(return_value=datetime(2030, 1, 1)))
+    @patch('util.config_life_cycle.ConfigParser.load_file', MagicMock(return_value=parsed_config))
+    @patch('util.config_life_cycle.ConfigLifeCycle.get_baking_cfg_file', MagicMock(return_value=""))
+    def test_wrong_reward_provider(self):
+        # Test with PRPC node
+        args = Args(initial_cycle=10, reward_data_provider='tzkt33', api_base_url='https://api.carthage.tzkt.io/v1/')
+        args.node_endpoint = 'https://testnet-tezos.giganode.io:443'
+        args.docker = True
+        args.dry_run = True
+        args.dry_run_no_consumers = True
+        args.syslog = False
+        args.verbose = "off"
+        args.log_file = 'logs/app.log'
+        args.do_not_publish_stats = True
+        args.run_mode = 3
+
+        start_application(args)
+        pass
+
+    @patch('util.process_life_cycle.verbose_logger', test_logger)
+    @patch('log_config.verbose_logger', test_logger)
+    @patch('log_config.init', MagicMock())
+    @patch('cli.client_manager.ClientManager.check_pkh_known_by_signer', MagicMock(return_value=True))
+    @patch('cli.client_manager.ClientManager.get_bootstrapped', MagicMock(return_value=datetime(2030, 1, 1)))
+    @patch('util.config_life_cycle.ConfigParser.load_file', MagicMock(return_value=parsed_config))
+    @patch('util.config_life_cycle.ConfigLifeCycle.get_baking_cfg_file', MagicMock(return_value=""))
+    def test_wrong_args_run_mode(self):
+        # Test with PRPC node
+        args = Args(initial_cycle=10, reward_data_provider='tzkt33', api_base_url='https://api.carthage.tzkt.io/v1/')
+        args.node_endpoint = 'https://testnet-tezos.giganode.io:443'
+        args.docker = True
+        args.dry_run = True
+        args.dry_run_no_consumers = True
+        args.syslog = False
+        args.verbose = "off"
+        args.log_file = 'logs/app.log'
+        args.do_not_publish_stats = True
+        args.run_mode = 33
+
+        start_application(args)
+        pass
```

Comment:
see above

Reply:
Can you change those parts yourself and add your work amount please? I will merge then